pkg/cli/image/extract/extract.go

```diff
@@ -375,17 +377,31 @@ func (o *ExtractOptions) Run() error {
 					return fmt.Errorf("unable to connect to image repository %s: %v", from.String(), err)
 				}
 
-				srcManifest, location, err := imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
+				srcManifest, location, err := retrieveSourceManifest(ctx, from, repo, o)
 				if err != nil {
-					if imagemanifest.IsImageForbidden(err) {
-						msg := fmt.Sprintf("image %q does not exist or you don't have permission to access the repository", from)
-						return imagemanifest.NewImageForbidden(msg, err)
+					if err != imagemanifest.AllImageFilteredErr {
+						return err
 					}
-					if imagemanifest.IsImageNotFound(err) {
-						msg := fmt.Sprintf("image %q not found: %s", from, err.Error())
-						return imagemanifest.NewImageNotFound(msg, err)
+
+					// Translate the current runtime OS to linux when looking through a manifest listed image since a manifest for the particular 'runtime OS'/'archType' may not exist.
+					// In cases where the runtime OS is windows or darwin (i.e. macOS), there is no expectation of a manifest image with those particular OS's.
+					if o.FilterOptions.DefaultOSFilter {
+						runtimeOS := runtime.GOOS
+						arch := runtime.GOARCH
+						if !warnedAboutLinuxConversion {
+							warnedAboutLinuxConversion = true
+							klog.Warningf("Warning: a manifest image could not be found for this platform: %s/%s. Converting runtime OS, %s, to 'linux' to pull the linux/%s equivalent image.\n", runtimeOS, arch, runtimeOS, arch)
```

Comment:
After investigating this further;

It would be better to remove `warnedAboutLinuxConversion` flag altogether and logging this in any case with V(2) like this

```go
klog.V(2).Infof("a manifest image could not be found for this platform: %s/%s. Converting runtime OS, %s, to 'linux' to pull the linux/%s equivalent image.\n", runtimeOS, arch, runtimeOS, arch)
```

WDYT?

Reply:
I guess this would be okay since it's hidden in the verbose level. It still repeats there:
```
brcox@brcox-mac oc % ./oc adm release extract --command=oc quay.io/openshift-release-dev/ocp-release:4.12.8-multi --to=/tmp/ -a ~/all-the-pull-secrets.json -v 2
I0329 06:50:01.736430   59066 extract.go:393] Warning: a manifest image could not be found for this platform: %s/%s. Converting runtime OS, %s, to 'linux' to pull the linux/%s equivalent image.
darwinamd64darwinamd64
I0329 06:50:02.013744   59066 manifest.go:451] Chose linux/amd64 manifest from the manifest list.
I0329 06:50:03.377163   59066 extract_tools.go:479] Will extract usr/share/openshift/mac/oc from quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:8653f3d359970b5126f6ada6618898a61da9ae7ff88dbcbec59eed249dc85765
I0329 06:50:03.377555   59066 extract_tools.go:466] Skipping openshift-client-mac-arm64-%s.tar.gz, does not match current architecture arm64
I0329 06:50:03.377917   59066 extract_tools.go:461] Skipping openshift-client-linux-%s.tar.gz, does not match current OS linux
I0329 06:50:03.377936   59066 extract_tools.go:461] Skipping openshift-client-linux-amd64-%s.tar.gz, does not match current OS linux
I0329 06:50:03.377950   59066 extract_tools.go:461] Skipping openshift-client-linux-arm64-%s.tar.gz, does not match current OS linux
I0329 06:50:03.377962   59066 extract_tools.go:461] Skipping openshift-client-windows-%s.zip, does not match current OS windows
I0329 06:50:03.651705   59066 extract.go:393] Warning: a manifest image could not be found for this platform: %s/%s. Converting runtime OS, %s, to 'linux' to pull the linux/%s equivalent image.
darwinamd64darwinamd64
I0329 06:50:03.821987   59066 manifest.go:451] Chose linux/amd64 manifest from the manifest list.
I0329 06:53:02.564941   59066 extract_tools.go:853] Found match for release version at 3832 (len=4096, offset=93, n=4003)
```

Reply:
If you are good with that, I will push the changes up.